sql/src/main/java/io/crate/analyze/relations/RelationNormalizer.java

```diff
@@ -281,63 +282,50 @@ public Boolean visitFunction(Function symbol, Void context) {
 
         @Override
         public Boolean visitField(Field field, Void context) {
-            return FIELD_RELATION_VISITOR.process(field.relation(), field, false);
+            Boolean result = FIELD_RELATION_VISITOR.process(field.relation(), field);
+            return (result != null) && result;
         }
     }
 
     /**
      * Visits an output symbol in a queried relation using the provided field index.
      */
-    private static class FieldRelationVisitor<R> extends AnalyzedRelationVisitor<FieldRelationVisitorContext<R>, R> {
+    private static class FieldRelationVisitor<R> extends AnalyzedRelationVisitor<Field, R> {
```

Comment:
I don't see how we could use Void here. We need the result for this visitor on `AggregateFunctionReferenceVisitor` to be a Boolean and on `FieldReplacingVisitor` to be a Symbol.